src/backend/distributed/executor/multi_utility.c

```diff
@@ -816,6 +832,131 @@ ProcessAlterObjectSchemaStmt(AlterObjectSchemaStmt *alterObjectSchemaStmt,
 }
 
 
+/* TODO: Write function comments */
+static Node *
+ProcessVacuumStmt(VacuumStmt *vacuumStmt, const char *vacuumCommand, bool isTopLevel)
+{
+	Oid relationId = InvalidOid;
+	List *taskList = NIL;
```

Comment:
Agreed; I also wanted to run the master `VACUUM` first, but doing so was in opposition to the existing structure. Since you want to do the same, I'm changing it.